### PR TITLE
Added force_http_version_10 option

### PR DESCRIPTION
**Pull request by dantleech**
This PR forces the HTTP version 1.0 mode on the curl library.

We encountered the same issue reported in #89 but changing the curl version has not helped.

/cc @chirimoya

### CODE REVIEW

File: src/Jackalope/RepositoryFactoryJackrabbit.php [hunk 1]
```diff
@@ -93,6 +93,9 @@ public function getRepository(array $parameters = null)
         if (isset($parameters['jackalope.check_login_on_server'])) {
             $transport->setCheckLoginOnServer($parameters['jackalope.check_login_on_server']);
         }
+        if (isset($parameters['jackalope.force_http_version_10'])) {
+            $transport->forceHttpVersion10($parameters['jackalope.force_http_version_10']);
+        }
```
Comment (dbu) [inline src/Jackalope/RepositoryFactoryJackrabbit.php]: afaik you also need to allow this parameter https://github.com/jackalope/jackalope-jackrabbit/blob/force_http_10/src/Jackalope/RepositoryFactoryJackrabbit.php#L37

File: src/Jackalope/Transport/Jackrabbit/Request.php [hunk 2]
```diff
@@ -463,6 +477,10 @@ protected function singleRequest($getCurlObject)
         $curl->setopt(CURLOPT_URL, reset($this->uri));
         $curl->setopt(CURLOPT_HTTPHEADER, $headers);
         $curl->setopt(CURLOPT_POSTFIELDS, $this->body);
+
+        if (true === $this->forceHttpVersion10) {
+            $curl->setopt(CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_0);
```
Comment (dantleech) [inline src/Jackalope/Transport/Jackrabbit/Request.php]: I was going to add this to the `multiRequest` method, however I noticed that it is not called at anypoint during the test suite.

**Comment by dbu** [comment]
ah, interesting find. can you try to catch that chunked exception and enhance it to give a hint to the user that he might need to activate this option to avoid the error? otherwise people will not know they need to do that...

**Comment by dantleech** [comment]
Updated. @chirimoya can you test?

**Comment by dantleech** [comment]
@dbu I added and then removed the more specific exception - I am not sure in what other situations that exception might happen (its quite general "network recieving error"). I think if more people encouter this issue with the same error code, then we could add something more.

File: src/Jackalope/RepositoryFactoryJackrabbit.php [hunk 3]
```diff
@@ -42,6 +42,7 @@ class RepositoryFactoryJackrabbit implements RepositoryFactoryInterface
         'jackalope.disable_stream_wrapper' => 'boolean: if set and not empty, stream wrapper is disabled, otherwise the stream wrapper is enabled and streams are only fetched when reading from for the first time. If your code always uses all binary properties it reads, you can disable this for a small performance gain.',
         'jackalope.logger' => 'Psr\Log\LoggerInterface: Use the LoggingClient to wrap the default transport Client',
         Session::OPTION_AUTO_LASTMODIFIED => 'boolean: Whether to automatically update nodes having mix:lastModified. Defaults to true.',
+        'jackalope.force_http_version_10' => 'boolean: Force HTTP version 1.0, this can in solving problems with curl such as https://github.com/jackalope/jackalope-jackrabbit/issues/89',
```
Comment (dbu) [inline src/Jackalope/RepositoryFactoryJackrabbit.php]: i wonder if we should call this jackalope.jackrabbit_force_http_version_10 to go with jackalope.jackrabbit_expect as its a jackrabbit only option.

**Comment by dbu** [comment]
hm. i still think rewrapping the exception would be good, maybe just say something with little confidence like: "try setting the parameter ... to see if this solves the issue." 

i restarted the failing test with low requirements, lets see if it fails consistently.

File: src/Jackalope/RepositoryFactoryJackrabbit.php (hunk 3 above)
Comment (dantleech) [inline src/Jackalope/RepositoryFactoryJackrabbit.php]: +1

File: src/Jackalope/Transport/Jackrabbit/Request.php [hunk 4]
```diff
@@ -520,6 +539,12 @@ protected function handleError(curl $curl, $response, $httpCode)
             case CURLE_COULDNT_CONNECT:
                 $info = $curl->getinfo();
                 throw new NoSuchWorkspaceException($curl->error() . ' "' . $info['url'] . '"');
+            case CURLE_RECV_ERROR:
+                throw new RepositoryException(sprintf(
+                    'CURLE_RECV_ERROR (errno 56) encountered. This has been known to happen intermittently with ' .
+                    'some versions of libcurl (see https://github.com/jackalope/jackalope-jackrabbit/issues/89). ' .
+                    'You can use the "jackalope.jackrabbit_force_http_version_10" option to force HTTP 1.0 as a workaround'
```
Comment (dantleech) [inline src/Jackalope/Transport/Jackrabbit/Request.php]: @dbu sensible?

**Comment by dantleech** [comment]
Renamed parameter and added exception . /cc @chirimoya

File: src/Jackalope/Transport/Jackrabbit/Request.php (hunk 4 above)
Comment (dbu) [inline src/Jackalope/Transport/Jackrabbit/Request.php]: i like it. will wait for travis, then merge.

**Comment by dbu** [comment]
oh, can you also update the README? there we mention that topic too in the preconditions. i would just mention the config option as an alternative.

**Comment by dbu** [comment]
thanks dan! i was going to put my standard request of documenting this, but notice that we seem to have no reference of the configuration options in jackalope.github.io or the README of this repo.

we should however add the option to http://symfony.com/doc/current/cmf/bundles/phpcr_odm/configuration.html#phpcr-session-with-jackalope-jackrabbit (with a `since` information)

**Pull request merged**